dhall/src/Dhall/Eval.hs

```diff
@@ -132,873 +105,1198 @@ instance Monoid (VChunks a) where
-                            | otherwise -> error errorMsg
-                          (x, y, ma) -> VMerge x y ma
-    ToMap x ma       -> case (evalE x, evalE <$> ma) of
-                          (VRecordLit m, ma'@(Just _)) | null m ->
-                            VListLit ma' (Data.Sequence.empty)
-                          (VRecordLit m, _) -> let
-                            entry (k, v) =
-                              VRecordLit (Dhall.Map.fromList [("mapKey", VTextLit $ VChunks [] k),
-                                                              ("mapValue", v)])
-                            s = (Data.Sequence.fromList . map entry . Dhall.Map.toList) m
-                            in VListLit Nothing s
-                          (x, ma) -> VToMap x ma
-    Field t k        -> vField (evalE t) k
-    Project t (Left ks) -> vProjectByFields env (evalE t) (Dhall.Set.sort ks)
-    Project t (Right e) ->
-                        case evalE e of
-                          VRecord kts ->
-                            evalE (Project t (Left (Dhall.Set.fromSet (Dhall.Map.keysSet kts))))
-                          e' -> VProject (evalE t) (Right e')
-    Assert t         -> VAssert (evalE t)
-    Equivalent t u   -> VEquivalent (evalE t) (evalE u)
-    Note _ e         -> evalE e
-    ImportAlt t _    -> evalE t
-    Embed a          -> VEmbed a
-
-
--- Conversion checking
---------------------------------------------------------------------------------
+    evalChunks (Chunks xys z) = foldr' cons nil xys
+      where
+        cons (x, t) vcs =
+            case evalE t of
+                VTextLit vcs' -> VChunks [] x <> vcs' <> vcs
+                t             -> VChunks [(x, t)] mempty <> vcs
+
+        nil = VChunks [] z
+    {-# INLINE evalChunks #-}
 
 eqListBy :: (a -> a -> Bool) -> [a] -> [a] -> Bool
-eqListBy f = go where
-  go (x:xs) (y:ys) | f x y = go xs ys
-  go [] [] = True
-  go _  _  = False
-{-# inline eqListBy #-}
+eqListBy f = go
+  where
+    go (x:xs) (y:ys) | f x y = go xs ys
+    go [] [] = True
+    go _  _  = False
+{-# INLINE eqListBy #-}
 
 eqMapsBy :: Ord k => (v -> v -> Bool) -> Map k v -> Map k v -> Bool
 eqMapsBy f mL mR =
-    Dhall.Map.size mL == Dhall.Map.size mR
-    && eqListBy eq (Dhall.Map.toList mL) (Dhall.Map.toList mR)
+    Map.size mL == Map.size mR
+    && eqListBy eq (Map.toList mL) (Map.toList mR)
   where
     eq (kL, vL) (kR, vR) = kL == kR && f vL vR
-{-# inline eqMapsBy #-}
+{-# INLINE eqMapsBy #-}
 
 eqMaybeBy :: (a -> a -> Bool) -> Maybe a -> Maybe a -> Bool
-eqMaybeBy f = go where
-  go (Just x) (Just y) = f x y
-  go Nothing  Nothing  = True
-  go _        _        = False
-{-# inline eqMaybeBy #-}
+eqMaybeBy f = go
+  where
+    go (Just x) (Just y) = f x y
+    go Nothing  Nothing  = True
+    go _        _        = False
+{-# INLINE eqMaybeBy #-}
 
-conv :: forall a. Eq a => Env a -> Val a -> Val a -> Bool
+conv :: forall a. Eq a => Environment a -> Val a -> Val a -> Bool
```

Comment:
Which term is `conv` the abbreviation for BTW?

Reply:
"conv" is short for "convergence checking"